### PR TITLE
8277980: ObjectMethods::bootstrap throws NPE when lookup is null

### PR DESCRIPTION
**Pull request by vicente-romero-oracle**
Please review this simple patch along with the related CSR. The `lookup` argument at method java.lang.runtime.ObjectMethods::bootstrap was ignored until the fix for [JDK-8261847](https://bugs.openjdk.java.net/browse/JDK-8261847) which improved the performance of the default implementation of Record::toString, given that now we are delegating part of the concatenation to StringConcatFactory::makeConcatWithConstants we do need now the lookup argument which can't be null anymore,

TIA

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed
- [x] Change requires a CSR request to be approved

### Issues
 * [JDK-8277980](https://bugs.openjdk.java.net/browse/JDK-8277980): ObjectMethods::bootstrap throws NPE when lookup is null
 * [JDK-8278316](https://bugs.openjdk.java.net/browse/JDK-8278316): ObjectMethods::bootstrap throws NPE when lookup is null (**CSR**)

### Reviewers
 * [Jonathan Gibbons](https://openjdk.java.net/census#jjg) (@jonathan-gibbons - **Reviewer**) ⚠️ Review applies to 6dbd35ca8ea6ce277c7aa037bcb6481b156a9a71

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/6754/head:pull/6754` \
`$ git checkout pull/6754`

Update a local copy of the PR: \
`$ git checkout pull/6754` \
`$ git pull https://git.openjdk.java.net/jdk pull/6754/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 6754`

View PR using the GUI difftool: \
`$ git pr show -t 6754`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/6754.diff">https://git.openjdk.java.net/jdk/pull/6754.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/java/lang/runtime/ObjectMethods.java [hunk 1]
```diff
@@ -396,15 +396,16 @@ private static MethodHandle makeToString(MethodHandles.Lookup lookup,
      *                     if invoked by a condy
      * @throws IllegalArgumentException if the bootstrap arguments are invalid
      *                                  or inconsistent
-     * @throws NullPointerException if any argument but {@code lookup} is {@code null},
-     *                              in the case of the {@code getters} argument, its
+     * @throws NullPointerException if any argument is {@code null}, in the case
+     *                              of the {@code getters} argument, its
```
Comment (jonathan-gibbons) [inline src/java.base/share/classes/java/lang/runtime/ObjectMethods.java:400]: OK. The second part about the contents of `getters` is a bit clunky, but was previously approved.

**Review by jonathan-gibbons — APPROVED** [review]
Looks OK, but given this is changing recently-reviewed code, I'd recommend getting a second opinion from one of the original reviewers, i.e. @mlchung

**Review by jonathan-gibbons — APPROVED** [review]

**Comment by vicente-romero-oracle** [comment]
I have updated the javadoc according to your comments

**Pull request closed without merging**